tests/functional/aws-node-sdk/test/bucket/getCorsResponse.js

```diff
@@ -0,0 +1,994 @@
+import assert from 'assert';
+import http from 'http';
+import https from 'https';
+
+import { S3 } from 'aws-sdk';
+
+import conf from '../../../../../lib/Config';
+import getConfig from '../support/config';
+
+const config = getConfig('default', { signatureVersion: 'v4' });
+const s3 = new S3(config);
+
+const transport = conf.https ? https : http;
+const bucket = 'bucketcorstester';
+const hostname = process.env.AWS_ON_AIR ? 's3.amazonaws.com' :
+    'localhost';
+
+const port = process.env.AWS_ON_AIR ? 80 : 8000;
+
+const statusCode = {
+    200: 200,
+    NoSuchBucket: 404,
+    BadRequest: 400,
+    AccessForbidden: 403,
+};
+
+// AWS seems to take a bit long so sometimes by the time we send the request
+// the bucket has not yet been created or the bucket has been deleted.
+function _waitForAWS(callback, err) {
+    if (process.env.AWS_ON_AIR) {
+        setTimeout((done, err) => {
+            callback(err);
+        }, 5000);
+    } else {
+        callback(err);
+    }
+}
+
+function methodRequest(method, headers, code, headersResponse, done,
+objectKey) {
+    const options = {
+        hostname,
+        port,
+        method,
+        headers,
+        path: objectKey ? `/${bucket}/${objectKey}` : `/${bucket}`,
+    };
+    const req = transport.request(options, res => {
+        const body = [];
+        res.on('data', chunk => {
+            body.push(chunk);
+        });
+        res.on('error', err => {
+            process.stdout.write('err on post response');
+            return done(err);
+        });
+        res.on('end', () => {
+            const total = body.join('');
+            if (code) {
+                const message = code === 200 ? '' : `<Code>${code}</Code>`;
+                assert(total.indexOf(message) > -1, `Expected ${message}`);
```

Comment:
I think it's the relevant portion that we check for in our other tests (statusCode, code). This is an example of the "total" response:
`<?xml version="1.0" encoding="UTF-8"?><Error><Code>AccessForbidden</Code><Message>CORSResponse: This CORS request is not allowed. This is usually because the evalution of Origin, request method / Access-Control-Request-Method or Access-Control-Request-Headers are not whitelisted by the resource's CORS spec.</Message><Resource></Resource><RequestId>73101446f9bcbc5b15ce</RequestId></Error>`

We can also check the `<Message>` element which happens to be the same at the moment, but that could change. The RequestId will also differ.